### PR TITLE
[AIP-132] Check unknown fields on list responses.

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: docs/rules/0132/response-unknown-fields.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+# List methods: Unknown fields (Response)
+
+This rule enforces that all `List` standard methods do not have unexpected
+fields, as mandated in [AIP-132][].
+
+## Details
+
+This rule looks at any message matching `List*Request` and complains if it
```
Comment (mingzhi) [inline docs/rules/0132/response-unknown-fields.md]: Is "List*Request" a typo?

File: rules/aip0132/response_unknown_fields.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// The resource itself is not included here, but also permitted.
+// This is covered in code in the rule itself.
+var respAllowedFields = stringset.New(
+	"next_page_token",       // AIP-158
+	"total_size",            // AIP-132
+	"unreachable",           // AIP-217
+	"unreachable_locations", // Wrong, but a separate AIP-217 rule catches it.
+)
+
+var responseUnknownFields = &lint.MessageRule{
```
Comment (mingzhi) [inline rules/aip0132/response_unknown_fields.go]: Can you use a `FieldRule`? You can add a `OnlyIf` function like:
```golang
func (f *desc.FieldDescriptor) bool {
  return isListResponseMessage(f.GetOwner())
}
```

File: docs/rules/0132/response-unknown-fields.md (hunk 1 above)
Comment (lukesneeringer) [reply docs/rules/0132/response-unknown-fields.md]: Yes. Thanks.

```suggestion
This rule looks at any message matching `List*Response` and complains if it
```

File: rules/aip0132/response_unknown_fields.go (hunk 2 above)
Comment (lukesneeringer) [reply rules/aip0132/response_unknown_fields.go]: I can, but does it really matter?

Comment (lukesneeringer) [reply rules/aip0132/response_unknown_fields.go]: I asked this before reading your refactor PR. In the context of _that_ PR, it obviously does matter.

Comment (mingzhi) [reply rules/aip0132/response_unknown_fields.go]: We can leave it as-is. Will refactor it later.

Comment (lukesneeringer) [reply rules/aip0132/response_unknown_fields.go]: Actually I am happy to change it.

Comment (lukesneeringer) [reply rules/aip0132/response_unknown_fields.go]: Done.

**Pull request merged**